src/main/kotlin/no/nav/syfo/service/PapirSmRegistreringMapperService.kt

```diff
@@ -253,14 +253,14 @@ private fun toPrognose(prognose: PrognoseType?): Prognose? = Prognose(
 
 private fun toArbeidsgiver(arbeidsgiver: ArbeidsgiverType?): Arbeidsgiver? = Arbeidsgiver(
         navn = arbeidsgiver?.navnArbeidsgiver,
-        harArbeidsgiver = when (arbeidsgiver?.harArbeidsgiver) {
-
-            // Copied from FellesformatMapperService.kt
-            "Flere arbeidsgivere" -> HarArbeidsgiver.FLERE_ARBEIDSGIVERE
-            "Flerearbeidsgivere" -> HarArbeidsgiver.FLERE_ARBEIDSGIVERE
-            "En arbeidsgiver" -> HarArbeidsgiver.EN_ARBEIDSGIVER
-            "Ingen arbeidsgiver" -> HarArbeidsgiver.INGEN_ARBEIDSGIVER
-            else -> HarArbeidsgiver.EN_ARBEIDSGIVER
+        harArbeidsgiver = with (arbeidsgiver?.harArbeidsgiver?.toLowerCase()) {
```

Comment:
jeg skjønner ikke helt hvorfor du må bruke with her? Synes koden var en del ryddigere før :/

Reply:
Enig, men den tidligere koden funket ikke... 

Tidligere sjekket jeg hele strengen. Problemet er at vi, ikke helt vet hvordan input ser ut, og vi har heller ingen måte å begrense det på. Nå fikk vi skanningdata hvor verdien i feltet bare var "En". Inputfeltet er bare en tekststreng, så selv om den forrige måten å sjekke på er ryddigere, så tror jeg det blir litt mange tilfeller å sjekke hvis vi skal ta for oss alle potensielle inputverdier...

Men det er kanskje mulig å gjøre det på en penere måte, uten with? Jeg er ikke helt stødig i Kotlin enda, så det kan fort hende! :)

Reply:
Hmm, skjønner.. Prøvde litt nå, men det var vrient å få det veldig pent dette her altså.. Det beste jeg fikk til var dette: 
`    harArbeidsgiver = when {
        arbeidsgiver?.harArbeidsgiver?.toLowerCase()?.contains("ingen") == true -> HarArbeidsgiver.INGEN_ARBEIDSGIVER
        arbeidsgiver?.harArbeidsgiver?.toLowerCase()?.contains("en") == true -> HarArbeidsgiver.EN_ARBEIDSGIVER
        arbeidsgiver?.harArbeidsgiver?.toLowerCase()?.contains("flere") == true -> HarArbeidsgiver.FLERE_ARBEIDSGIVERE
        !arbeidsgiver?.harArbeidsgiver.isNullOrEmpty() -> HarArbeidsgiver.EN_ARBEIDSGIVER
        else -> HarArbeidsgiver.INGEN_ARBEIDSGIVER
    }`

Men det er jo ikke superelegant det heller kanskje... Men man slipper all nullsjekkingen i det minste

Reply:
Visningen til github gjorde det ikke bedre akkurat, men det ser litt finere ut hvis man ser på det i intellij enn her i kommentaren :P 

Forresten, viktig å matche på "ingen" før "en" siden "ingen" inneholder "en" 😅

Reply:
Ufattelig godt poeng ang. "ingen" før "en"! Den kunne blitt stygg :p 

Ikke helt sikker på om jeg liker all den gjentakende bruken av toLowerCase() heller... Jeg prøver litt til :)

Reply:
Pushet ny kode nå :)